### DISCUSSION TITLE
[$50 reward] Livewire with third party JS plugin

### DISCUSSION THREAD

**Discussion by harryqt** [livewire/livewire#4578] [1. Help] [answered]
Check the repo for all info: https://github.com/Dibbyo456/livewire-help

**Comment by Lupennat** [comment, marked as answer]
@Dibbyo456 

probably should work by adding

```html
...
<script>
        document.addEventListener("DOMContentLoaded", () => {
            Livewire.hook('message.processed', (message, component) => {
                $('[data-toggle="table"]').bootstrapTable();
            })
        });
</script>
```

on the layout before closing body tag

**Reply by harryqt** [reply]
Nope. Did you mean `element.updated`? Tried both did not work. Its actually much harder than looks. :(

**Reply by Lupennat** [reply]
@Dibbyo456 take a look at the pull request, it should works

**Comment by vlados** [comment]
The problem is not Livewire, it's the bootstrap table. To be working it needs to destroy the table first and then create it:
```js
    document.addEventListener("DOMContentLoaded", () => {
        Livewire.hook('element.updating', (el, component) => {
            $('#table').bootstrapTable("destroy");
        })
        Livewire.hook('message.processed', (message, component) => {
            $('#table').bootstrapTable()
        })
    });
```
You can use packages like
https://github.com/rappasoft/laravel-livewire-tables 
https://laravelviews.com/table-view?sortOrder=asc
